src/content/mod_configuration.cpp

```diff
@@ -230,18 +230,34 @@ void ModConfiguration::resolveDependencies()
 	// of each mod, split mods into satisfied and unsatisfied
 	std::vector<ModSpec> satisfied;
 	std::list<ModSpec> unsatisfied;
+	ModSpec firstModSpec;
 	for (ModSpec mod : m_unsatisfied_mods) {
+		// dependencies are not allowed for first ad last mod
+		if ( (mod.name == first_mod) || (mod.name == last_mod) ) {
+			if (!mod.depends.empty() || !mod.optdepends.empty())
+				throw ModError("Mod selected as first/last by game is not allowed to have dependencies.");
+		}
 		mod.unsatisfied_depends = mod.depends;
 		// check which optional dependencies actually exist
 		for (const std::string &optdep : mod.optdepends) {
 			if (modnames.count(optdep) != 0)
 				mod.unsatisfied_depends.insert(optdep);
 		}
 		// if a mod has no depends it is initially satisfied
-		if (mod.unsatisfied_depends.empty())
-			satisfied.push_back(mod);
-		else
+		if (mod.unsatisfied_depends.empty()) {
```

Comment:
Doing this in this loop is all a bit messy. It is not obvious why it works. Why not just:

* Prepend the "first" mod, if existent, to the mod load order
* Append the "last" mod, if existent, to the mod load order
* Ignore the "first" and "last" mod in this loop

(Also make sure to check that (1) nothing depends on the last mod and (2) the first mod depends on nothing.)

Reply:
Fixed.